alacritty_terminal/src/display.rs

```diff
@@ -224,6 +224,44 @@ impl Display {
             api.clear(background_color);
         });
 
+        // We should call `clear` when window is offscreen, so when `window.show()` happens it
+        // would be already with background color instead of bunch of borders + uninitialized
+        // surface.
+        window.swap_buffers()?;
+
+        window.show();
+
+        // Set window position
+        //
+        // TODO: replace `set_position` with `with_position` once available
+        // Upstream issue: https://github.com/tomaka/winit/issues/806
+        if let Some(position) = config.window.position {
+            let physical = PhysicalPosition::from((position.x, position.y));
+            let logical = physical.to_logical(window.hidpi_factor());
+            window.set_position(logical);
+        }
+
+        let startup_mode = config.window.startup_mode();
+        if StartupMode::Fullscreen == startup_mode {
+            window.set_fullscreen(true);
+        }
+
+        #[cfg(target_os = "macos")]
+        {
+            if StartupMode::SimpleFullscreen == startup_mode {
+                use glutin::os::macos::WindowExt;
+                window.set_simple_fullscreen(true);
+            }
+        }
+
+
+        #[cfg(not(any(target_os = "macos", target_os = "windows")))]
+        {
+            if window.x11_window() && StartupMode::Maximized == startup_mode {
+                window.set_maximized(true);
+            }
+        }
```

Comment:
Looking at the three `startup_mode` statements here, I feel like it would be great to just put this into a `match` statement. That would clean up the code a lot.

This isn't related to this PR at all, but I usually think that if there's a chance to refactor things a bit, it's probably good to take it. Do you want to implement that or should I send a separate PR?

I'm thinking about something like this:

```rust
match config.window.startup_mode() {
    Fullscreen => window.set_fullscreen(true),
    #[cfg(target_os = "macos")]
    SimpleFullscreen => window.set_simple_fullscreen(true),
    #[...]
    Maximized if window.x11_window() => window.set_maximized(true),
    _ => (),
}
```

I also think the `use glutin::os::macos::WindowExt` shouldn't be required here, so that can probably be removed while you're at it.

Reply:
Yes, sure.